docs/modules/ROOT/pages/headers.adoc

```diff
@@ -0,0 +1,83 @@
+# DataSonnet Header
+
+The header is an optional comment section in the mapping document which must start with `/\** DataSonnet` followed by a new line character and end with a line `*/` and contains a set of configuration properties.
+The properties are hierarchical, with sections separated by the `.` (dot) character, e.g.:
```

Comment:
One of the comments that was removed with changes is pretty important: we need to add a restriction to what can be the parameters for a plugin, because now it needs to be restricted to what will load from the header.

Reply:
Well, there is a restriction, if a parameter is not recognized, an exception will be thrown by the plugin.

Reply:
One can express a list, e.g. input.*.test/test.TestParam=a,b,c
Of course it will be passed to the plugin as a string, and plugin itself can parse and handle it in any way. I say adding types at this point overcomplicates things, it will always be passed as a String and if plugin expects a number and the string does not parse as a number, the plugin itself will throw an exception.